### PR TITLE
[stdlib][fix] Fixed wrong sorting behavior

### PR DESCRIPTION
**Pull request by kchibisov**
Fix for this [bug](https://bugs.swift.org/browse/SR-8682).

We were using base index instead of index from current iteration in _siftDown.

### CODE REVIEW

File: stdlib/public/core/Sort.swift [hunk 1]
```diff
@@ -507,7 +507,7 @@ extension MutableCollection where Self: RandomAccessCollection {
       // If a child is bigger than the current node, swap them and continue sifting
       // down.
       if largest != i {
-        swapAt(idx, largest)
+        swapAt(i, largest)
```
Comment (xwu) [inline stdlib/public/core/Sort.swift]: If there is no reason ever to use the original value `idx`, then the first line of the function body should be `var idx = idx`.

**Review by xwu — COMMENTED** [review]
Glad the issue could be tracked down so quickly.

We need tests. At minimum, the original example in the bug report. But this raises concern for large gaps in testing and ideally a more systematic approach to closing them would be important.

**Comment by airspeedswift** [comment]
@swift-ci please test

**Comment by airspeedswift** [comment]
@swift-ci please smoke benchmark

**Comment by airspeedswift** [comment]
Merging to fix bug on mater, but we may need to add further tests on top of what's here as a follow-on.

**Comment by airspeedswift** [comment]
On the bright side, this restores my hope that we can have a stable sort that doesn't significantly lose out to this one in terms of performance :)

**Pull request merged**